src/bot/integrations/discord.ts

```diff
@@ -371,6 +394,28 @@ class Discord extends Integration {
     } catch (e) {
       warning(e.stack);
     }
+
+    this.changeClientOnlinePresence();
+  }
+
+  @onChange('onlinePresence')
+  async changeClientOnlinePresence() {
+    try {
+      if (api.isStreamOnline) {
+        if (this.togglePresenceOnStream === 'streaming') {
+          this.client?.user?.setStatus('online');
+          this.client?.user?.setPresence({ status: 'online', activity: { name: `${api.stats.currentTitle}`, type: 'STREAMING', url: `https://twitch.tv/${oauth.generalChannel}`} });
+        } else {
+          this.client?.user?.setStatus(this.togglePresenceOnStream);
+          this.client?.user?.setPresence({ status: this.togglePresenceOnStream, activity: { name: '', type: undefined, url: '' } });
```

Comment:
Also I am not super happy with empty activity, can it be customized by settings? what can be in name? type? url? We can probably add more variables to this, couldn't we?

Reply:
name can be any string
type [ActivityType](https://discord.js.org/#/docs/main/stable/typedef/ActivityType)
url is twitch link ONLY

Reply:
if activity can be any string, we can have it configurable same as status. if you use 

```
const activityString = await new Message(this.offlineActivityName).parse({});
```

than we can use all filters

Reply:
`the setPresence( activity{name: activityString} )`
can only exist if type is set to streaming/watching/listening
a discord bot cant use custom_status this is limited to normal user only
so its limited in usecase

but having online activity name config is a idea
and a toggle for onHost --> `setPresence( activity{name: $host type: 'WATCHING'} )`